iceoryx_examples/benchmark_optional_and_expected/CMakeLists.txt

```diff
@@ -0,0 +1,28 @@
+# Build icedelivery example
+cmake_minimum_required(VERSION 3.5)
+project(benchmark_optional_and_expected)
+
+include(GNUInstallDirs)
+
+find_package(iceoryx_utils CONFIG REQUIRED)
+
+get_target_property(ICEORYX_CXX_STANDARD iceoryx_utils::iceoryx_utils CXX_STANDARD)
+if ( NOT ICEORYX_CXX_STANDARD )
+    include(IceoryxPlatformDetection)
+endif ( NOT ICEORYX_CXX_STANDARD )
+
+add_executable(iox-benchmark-optional-and-expected ./benchmark_optional_and_expected.cpp)
+target_link_libraries(iox-benchmark-optional-and-expected
+    iceoryx_utils::iceoryx_utils
+    pthread
+)
+set_target_properties(iox-benchmark-optional-and-expected PROPERTIES 
+    CXX_STANDARD_REQUIRED ON
+    CXX_STANDARD ${ICEORYX_CXX_STANDARD}
+    POSITION_INDEPENDENT_CODE ON
+)
+
+install(
+    TARGETS iox-benchmark-optional-and-expected
```

Comment:
since I don't like too much typing, would `iox-bench-...` also be fine with you?

Reply:
Never! ;) But I could rename it into `iox-benchmark-binary-executable-from-iceoryx-to-test-optional-and-expected-with-output-statistics-oh-i-love-iceoryx-blaa`

Reply:
@elfenpiff I changed my mind. I want something like this
`iox-benchmark-elf-binary-executable-from-iceoryx-to-test-optional-and-expected-with-output-statistics-on-posix-operation-systems-like-linux-qnx-and-macos-and-additionally-windows-although-it-is-not-posix-to-compare-the-performace-impact-of-optional-and-expected-data-types-to-a-more-traditional-uses-of-bool-and-passing-references-to-functions-for-a-return-value`

Reply:
How about just `iox-bm-` ... keep with the two-letter convention we accidently ended up with with the gateway (`iox-gw-`) and examples (`iox-ex-`).

Reply:
Alrighty then `iox-bm-` it is